packages/wrangler/src/config/validation-pages.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Pages now supports configuration via `wrangler.toml`. As opposed to
+ * Workers however, Pages only supports a limited subset of all available
+ * configuration keys.
+ *
+ * This file contains all `wrangler.toml` validation things, specific to
+ * Pages.
+ */
+
+import { Diagnostics } from "./diagnostics";
+import { hasProperty } from "./validation-helpers";
+import type {
+	Config,
+	RawConfig,
+} from "./config";
+
+const supportedPagesConfigFields = [
```

Comment:
I am really hoping there's a better way and I'm too dumb to see it right now :/.

My original hope was we could use a dedicated type for this
```
export type PagesSupportedConfig = PagesConfigFields &
	DevConfig &
	Pick<Environment,
		"name" |
		"compatibility_date" |
		"compatibility_flags" |
		"placement" |
		"vars" |
		"kv_namespaces" |
		"queues" |
		"r2_buckets" |
		"d1_databases" |
		"vectorize" |
		"services" |
		"analytics_engine_datasets" |
		"ai"
	>;
	```
and use that to extract the keys. But extracting those as an array of strings is not supported in TS, so that was that. I eventually resorted to the good ol `Array`, but all years if TS gives us a more elegant way to achieve this